### PR TITLE
Adds account verification response to login flow

### PR DESCRIPTION
**Pull request by charliescheer**
### Fix
This PR adds a new error case while logging in where a user's email address has not been verified they will receive an error alerting them to the fact that they can not login till their email is verified and then supplies them with a button to resend the verification email.

### Test
1. setup a tool like Charles proxy to get in the way of the login request response.
2. When the response returns, change the response value to 403
3. You should see an alert with two options, verify and okay.

Note this will require more testing when the backend changes are made to return the 403 response to an unverified email, currently that has not been added in, so it can not be fully tested.

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 2.15
 -----
 -   [Internal] added login alert for compromised password #1017
+- [Internal] added error response case on login for unverified emails #1019
```
Comment (jleandroperez) [inline RELEASE-NOTES.txt:3]:
```suggestion
-   [Internal] added error response case on login for unverified emails #1019
```

Comment (jleandroperez) [reply RELEASE-NOTES.txt:3]: (Nipticky!! so that it has the exact same spacing as the line before)

File: Simplenote/AuthViewController.m [hunk 2]
```diff
@@ -353,6 +355,17 @@ -(void)presentPasswordCompromisedAlert
     }];
 }
 
+-(void) presentUnverifiedEmailAlert
```
Comment (jleandroperez) [inline Simplenote/AuthViewController.m]:
```suggestion
- (void)presentUnverifiedEmailAlert
```

File: Simplenote/AuthViewController+Swift.swift [hunk 3]
```diff
@@ -201,6 +201,22 @@ extension AuthViewController {
...
+        alert.informativeText = Localization.unverifiedInformativeText
+        alert.addButton(withTitle: Localization.unverifiedActionText)
+        alert.addButton(withTitle: Localization.unverifiedCancelText)
+
+        alert.beginSheetModal(for: window, completionHandler: completion)
+    }
+
+    @objc
+    func sendVerificationMessage(for email: String) {
+        AccountVerificationRemote().verify(email: email) { _ in }
```
Comment (jleandroperez) [inline Simplenote/AuthViewController+Swift.swift]: We should probably replicate what @eshurakov mentioned [here](https://github.com/Automattic/simplenote-ios/pull/1398#discussion_r690185403). Pasting for convenience!!

```
I think we should show an activity indicator (can use the one we have for account deletion) and handle errors similar to what we're doing in AccountVerificationViewController
```

**Review by jleandroperez — COMMENTED** [review]
Few notes!! most important one is the comment regarding Spinner / Error Handling.

TY!!

File: Simplenote/AuthViewController+Swift.swift (hunk 3 above)
Comment (charliescheer) [reply Simplenote/AuthViewController+Swift.swift]: Good idea.  I have added the check to confirm that the 403 error is "requires verification" and added a ProgressIndicator

File: Simplenote/AuthViewController+Swift.swift [hunk 4]
```diff
@@ -234,4 +264,12 @@ private enum Localization {
     static let compromisedPasswordMessage = NSLocalizedString("This password has appeared in a data breach, which puts your account at high risk of compromise. It is recommended that you change your password immediately.", comment: "Compromised password alert message")
     static let changePasswordAction = NSLocalizedString("Change Password", comment: "Change password action")
     static let dismissChangePasswordAction = NSLocalizedString("Not Now", comment: "Dismiss change password alert action")
+    static let unverifiedInformativeText = NSLocalizedString("You must verify your email before being able to login.", comment: "Erro for un verified email")
+    static let unverifiedMessageText = NSLocalizedString("Account Verification Required", comment: "Email verification required alert title")
+    static let cancelText = NSLocalizedString("Okay", comment: "Email unverified alert dismiss")
```
Comment (eshurakov) [inline Simplenote/AuthViewController+Swift.swift]: Similar comment as on iOS, we were always using "OK"

**Review by eshurakov — APPROVED** [review]
Looks good to me!

:shipit:

File: Simplenote/NSProgressIndicator+Simplenote.swift [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+extension NSProgressIndicator {
+    static func addProgressIndicator(to view: NSView) -> NSProgressIndicator {
+        let progressIndicator = NSProgressIndicator.init(frame: NSRect(x: view.frame.minX, y: view.frame.midY, width: 20, height: 20))
```
Comment (jleandroperez) [inline Simplenote/NSProgressIndicator+Simplenote.swift]:
```suggestion
        let progressIndicator = NSProgressIndicator(frame: NSRect(x: view.frame.minX, y: view.frame.midY, width: 20, height: 20))
```

Comment (jleandroperez) [reply Simplenote/NSProgressIndicator+Simplenote.swift]: Since you're using autolayout... can we try just running `NSProgressIndicator()` instead?

Comment (jleandroperez) [reply Simplenote/NSProgressIndicator+Simplenote.swift]: Plus let's also add `progressIndicator.controlSize = .small`, looks way cooler!

File: Simplenote/AuthViewController+Swift.swift [hunk 6]
```diff
@@ -202,6 +202,36 @@ extension AuthViewController {
...
+        alert.addButton(withTitle: Localization.unverifiedActionText)
+        alert.addButton(withTitle: Localization.cancelText)
+
+        alert.beginSheetModal(for: window, completionHandler: completion)
+    }
+
+    @objc
+    func sendVerificationMessage(for email: String, inWindow window: NSWindow) {
+        let progressIndicator = NSProgressIndicator.addProgressIndicator(to: view)
+
```
Comment (jleandroperez) [inline Simplenote/AuthViewController+Swift.swift:221]: Let's add `setInterfaceEnabled(false)` (and true on completion)

**Review by jleandroperez — APPROVED** [review]
Looks great!! few comments (re: Spinner size, locking the UI).

Other than that, :shipit: when ready!!

**Pull request merged**